src/unlink.js

```diff
@@ -13,60 +15,73 @@ const flatten = require('lodash').flatten;
 
 log.heading = 'rnpm-link';
 
+const unlinkDependencyAndroid = (androidProject, dependency, packageName) => {
+  if (!androidProject || !dependency.android) {
+    return;
+  }
+
+  const isInstalled = isInstalledAndroid(androidProject, packageName);
+
+  if (!isInstalled) {
+    log.info(`Android module ${packageName} is already unlinked`);
+    return;
+  }
+
+  log.info(`Unlinking ${packageName} android dependency`);
+
+  unregisterDependencyAndroid(packageName, dependency.android, androidProject);
+
+  log.info(`Android module ${packageName} has been successfully unlinked`);
+};
+
+const unlinkDependencyIOS = (iOSProject, dependency, packageName) => {
+  if (!iOSProject || !dependency.ios) {
+    return;
+  }
+
+  const isInstalled = isInstalledIOS(iOSProject, dependency.ios);
+
+  if (!isInstalled) {
+    log.info(`iOS module ${packageName} is already unlinked`);
+    return;
+  }
+
+  log.info(`Unlinking ${packageName} ios dependency`);
+
+  unregisterDependencyIOS(dependency.ios, iOSProject);
+
+  log.info(`iOS module ${packageName} has been successfully unlinked`);
+};
+
 /**
  * Updates project and unlink specific dependency
  *
  * If optional argument [packageName] is provided, it's the only one that's checked
  */
 module.exports = function unlink(config, args) {
+  const packageName = args[0];
+
+  var project;
+  var dependency;
+
   try {
-    const project = config.getProjectConfig();
+    project = config.getProjectConfig();
   } catch (err) {
     log.error('ERRPACKAGEJSON', `No package found. Are you sure it's a React Native project?`);
     return Promise.reject(err);
   }
 
-  const packageName = args[0];
-
   try {
-    const dependency = config.getDependencyConfig(packageName);
+    dependency = config.getDependencyConfig(packageName);
   } catch (err) {
-    log.warn(
-      'ERRINVALIDPROJ',
-      `Project ${packageName} is not a react-native library`
-    );
+    log.warn('ERRINVALIDPROJ', `Project ${packageName} is not a react-native library`);
```

Comment:
What I meant was I wasn't getting any warnings locally for this particular rule. I also thought we lint on CircleCi as well.